app/search/classification/tests/test_wikipedia.py

```diff
@@ -0,0 +1,110 @@
+from unittest import TestCase
+from unittest.mock import patch
+from urllib.parse import parse_qs, urlparse
+
+import responses
+from nose.tools import eq_, ok_
+
+from app.search.classification.wikipedia import WikipediaClassifier
+from app.tests.memcached import mock_memcached
+
+
+SLUG = 'Mozilla'
+TITLE = 'Mozilla'
+ABSTRACT = ' \n\tMozilla is a free-software community... \n\t'
+API_URL = ('http://example.com/')
+MOCK_RESPONSE = {
+    'query': {
+        'general': {},
+        'pages': {
+            '36754915': {
+                'extract': ABSTRACT,
+                'pageid': 36754915,
+                'title': TITLE,
+                'ns': 0
+            }
+        }
+    },
+    'batchcomplete': ''
+}
+EXPECTED_SANITIZED_RESPONSE = {
+    'pageid': 36754915,
+    'ns': 0,
+    'title': TITLE,
+    'extract': ABSTRACT
+}
+
+
+class TestWikipediaClassifier(TestCase):
+    def tearDown(self):
+        mock_memcached.flush_all()
+
+    def _result(self, url):
+        return {
+            'url': url
+        }
+
+    def _classifier(self, url):
+        return WikipediaClassifier(self._result(url))
+
+    def _matches(self, url):
+        return self._classifier(url).is_match(self._result(url))
+
+    def test_is_match(self):
+        eq_(self._matches('https://www.mozilla.com/en_US'), False)
+        eq_(self._matches('https://www.mozilla.com/en_US/'), False)
+        eq_(self._matches('https://wikipedia.org'), False)
+        eq_(self._matches('https://wikipedia.org/'), False)
+        eq_(self._matches('https://en.wikipedia.org/wiki/Mozilla'), True)
+        eq_(self._matches('https://en.wikipedia.org/wiki/Mozilla/'), True)
+
+    def test_api_url(self):
+        classifier = self._classifier('https://en.wikipedia.org/wiki/Mozilla')
+        expected = list(urlparse(
+            'https://en.wikipedia.org/w/api.php?exintro=&explaintext=&titles=M'
+            'ozilla&redirects=&prop=extracts&meta=siteinfo&action=query&format'
+            '=json'
+        ))
+        actual = list(urlparse(classifier._api_url('Mozilla')))
+        expected[4] = parse_qs(expected[4])
+        actual[4] = parse_qs(actual[4])
+        eq_(expected, actual)
+
+    @patch('app.memorize.memcached', mock_memcached)
+    @patch('app.search.classification.wikipedia.WikipediaClassifier._api_url')
+    @responses.activate
+    def test_api_response(self, mock_api_url):
+        """
+        Tests WikipediaClassifier._api_response by running it twice for the
+        same slug and:
+
+        * Asserting that the first is not returned from the cache, while the
+          second is.
+        * Asserting that the cache keys for each are the same.
+        * Asserting that the response body matches the expected sanitized
+          response.
+        """
```

Comment:
This feels a bit more like an integration test than a unit test, since we're verifying that the value returned from the (mock) cache matches the value returned from the (mock) api. Could it be useful to separate out integration tests from unit tests?